BasicSample/basic_sample.js

```diff
@@ -1,48 +1,120 @@
-handlers.helloWorld = function (args) {
-
-    // "currentPlayerId" is initialized to the PlayFab ID of the player logged-in on the game client. 
+// This is a Cloud Script function. "args" is set to the value of the "FunctionParameter" 
+// parameter of the ExecuteCloudScript API.
+// (https://api.playfab.com/Documentation/Client/method/ExecuteCloudScript)
+// "context" contains additional information when the Cloud Script function is called from a PlayStream action.
+handlers.helloWorld = function (args, context) {
+    
+    // The pre-defined "currentPlayerId" variable is initialized to the PlayFab ID of the player logged-in on the game client. 
     // Cloud Script handles authenticating the player automatically.
-    var message = "Hello " + currentPlayerId + "!";
+    var message = "Hello " + currentPlayerId + "! You said " + args.inputValue;
 
-    // You can use the "log" object to write out debugging statements. The "log" object has
-    // three functions corresponding to logging level: debug, info, and error.
+    // You can use the "log" object to write out debugging statements. It has
+    // three functions corresponding to logging level: debug, info, and error. These functions
+    // take a message string and an optional object.
     log.info(message);
-
-    // Whatever value you return from a CloudScript handler function is passed back 
-    // to the game client. It is set in the "Results" property of the object returned by the 
-    // RunCloudScript API. Any log statments generated by the handler function are also included 
-    // in the "ActionLog" field of the RunCloudScript result, so you can use them to assist in
-    // debugging and error handling.
+    log.debug("helloWorld:", { foo: "bar" });
+
+    // The value you return from a Cloud Script function is passed back 
+    // to the game client in the ExecuteCloudScript API response, along with any log statements
+    // and additional diagnostic information, such as any errors returned by API calls or external HTTP
+    // requests. They are also included in the optional player_executed_cloudscript PlayStream event 
+    // generated by the function execution.
+    // (https://api.playfab.com/playstream/docs/PlayStreamEventModels/player/player_executed_cloudscript)
     return { messageValue: message };
 }
 
+// This is a simple example of making a PlayFab server API call
+handlers.makeAPICall = function (args, context) {
+    
+    // The pre-defined "server" object has functions corresponding to each PlayFab server API 
+    // (https://api.playfab.com/Documentation/Server). It is automatically 
+    // authenticated as your title and handles all communication with 
+    // the PlayFab API, so you don't have to write extra code to issue HTTP requests. 
+	var playerStatResult = server.UpdateUserStatistics (
+		{
+			PlayFabId: currentPlayerId,
+			UserStatistics: {Level:2}
+		}
+	);
+}
+
+// This is a simple example of making a web request to an external HTTP API.
```

Comment:
It should be noted that this makeHTTPRequest only supports "Content-Type: application/json", and that there are other formats available, which probably work, but aren't explicitly demo'd.  If we want to explicitly support those other protocols, we should probably have more examples.